packages/dev/core/src/FlowGraph/flowGraphPath.ts

```diff
@@ -0,0 +1,155 @@
+            });
+        }
+        return { pathParts, templateStrings };
+    }
+
+    /**
+     * Gets the template strings in this path.
+     * @returns an array containing the template strings in this path.
+     */
+    getTemplateStrings(): string[] {
+        return this._templateStrings;
+    }
+
+    setTemplateSubstitution(template: string, value: number) {
+        if (this._templateStrings.indexOf(template) === -1) {
+            throw new Error(`Template string ${template} does not exist in path ${this._path}`);
+        }
+        this._templateSubstitutions[template] = value;
+    }
+
+    private _evaluateTemplates() {
+        for (const pathPart of this._pathParts) {
+            if (pathPart.isTemplate) {
+                const value = this._templateSubstitutions[pathPart.valueWithoutBraces];
+                if (value === undefined) {
+                    throw new Error(`Template string ${pathPart.value} was not substituted`);
+                }
+                pathPart.replacedValue = value.toString();
+            }
+        }
+    }
+
+    private _getFinalPath() {
+        let finalPath = "";
+        for (const pathPart of this._pathParts) {
+            if (pathPart.isTemplate) {
+                finalPath += pathPart.replacedValue;
+            } else {
+                finalPath += pathPart.value;
+            }
+        }
+        return finalPath;
+    }
+
+    /*
+     * Breaks the path into a chain of entities, for example,
+     * /x/y/z would be split into [context._userVariables.x, context._userVariables.x.y, context._userVariables.x.y.z]
+     */
+    private _evaluatePath(context: FlowGraphContext): any {
+        this._evaluateTemplates();
+
+        const entityChain = [];
+        const splitPath = [];
+        let currentTarget = context._userVariables;
+        for (const pathPart of this._pathParts) {
+            if (currentTarget === undefined) {
+                throw new Error(`Could not find path ${this._getFinalPath()} in target context`);
+            }
+            const value = pathPart.isTemplate ? pathPart.replacedValue : pathPart.value;
+            if (!value) {
+                throw new Error(`Invalid path ${this._getFinalPath()}`);
+            }
+            currentTarget = currentTarget[value];
+            entityChain.push(currentTarget);
+            splitPath.push(value);
+        }
+
+        return { entityChain, splitPath };
+    }
+
+    getProperty(context: FlowGraphContext): any {
+        const { entityChain } = this._evaluatePath(context);
+        return entityChain[entityChain.length - 1];
+    }
+
+    setProperty(context: FlowGraphContext, value: any) {
+        const { entityChain, splitPath } = this._evaluatePath(context);
+        const target = entityChain[entityChain.length - 2];
+        const property = splitPath[splitPath.length - 1];
+        target[property] = value;
+    }
+
+    getClassName() {
+        return "FGPath";

```

Comment:
```suggestion
        return "FlowGraphPath";
```

Reply:
I'm using "FG" as the prefix to stay consistent with the other Flow Graph block parts, where I'm also using "FG" to get shorter class names.